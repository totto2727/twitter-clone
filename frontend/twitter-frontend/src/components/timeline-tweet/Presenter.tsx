const Presenter = () => {
  return (
    <div className={'px-4 py-3 border-t border-b flex gap-3'}>
      <div>icon</div>
      <div>
        <div className={'flex gap-2'}>
          <span>name</span>
          <span>id</span>
        </div>
        <div>text</div>
        <div>image</div>
        <div>movie</div>
        <div>card</div>
        <div>このスレッドを表示</div>
      </div>
    </div>
  )
}
export default Presenter
