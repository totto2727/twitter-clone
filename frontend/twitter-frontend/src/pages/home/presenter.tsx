const Presenter: React.FC = () => {
  return (
    <main>
      <div>
        <p>Home</p>
      </div>
      <div>
        <p>input</p>
        <p>navi</p>
      </div>
      <article>
        <div>
          <p>timeline</p>
        </div>
      </article>
    </main>
  )
}

export default Presenter
