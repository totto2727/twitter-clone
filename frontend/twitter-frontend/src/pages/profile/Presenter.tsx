export type Props = {
  userId: number
}

const Presenter: React.VFC<Props> = ({ userId }) => {
  return (
    <>
      <h1>{userId}</h1>
    </>
  )
}

export default Presenter
