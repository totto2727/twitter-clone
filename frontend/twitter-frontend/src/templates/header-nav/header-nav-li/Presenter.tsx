import IconLink from '@/icon-link'
import { IconLinkProps } from '@/icon-link/Presenter'

export type HeaderNavLiProps = Omit<IconLinkProps, 'children'> & {
  current?: boolean
  text?: string
}

const Presenter = ({
  className,
  text,
  iconProps,
  current,
  ...props
}: HeaderNavLiProps) => {
  const props_: IconLinkProps = {
    className:
      'p-3 flex xl:inline-flex items-center justify-center xl:justify-start gap-8 border transition hover:bg-white hover:bg-opacity-10 duration-300',
    iconProps: {
      size: 26,
      fill: current ?'white':undefined,
      strokeWidth: current ? 20 : 0,
      ...iconProps,
    },
    ...props,
  }
  return (
    <li className={className}>
      <IconLink {...props_}>
        {text && <p className='hidden xl:block text-xl border'>{text}</p>}
      </IconLink>
    </li>
  )
}

export default Presenter
