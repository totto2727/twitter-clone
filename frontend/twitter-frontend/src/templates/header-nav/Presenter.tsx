import HeaderNavLi from './header-nav-li'
import { HeaderNavLiProps } from './header-nav-li/Presenter'

export type HeaderNavProps = {
  links: HeaderNavLiProps[]
}

const Presenter = ({ links }: HeaderNavProps) => {
  return (
    <nav className={'md:mx-4 xs:h-full flex-none  xs:w-16 xl:w-64'}>
      <ul
        className={'flex flex-row justify-around xs:flex-col xs:justify-start'}>
        {links.map((p, i) => (
          <HeaderNavLi {...p} key={i} />
        ))}
      </ul>
    </nav>
  )
}

export default Presenter
