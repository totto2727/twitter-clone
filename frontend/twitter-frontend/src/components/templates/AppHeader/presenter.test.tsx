import renderer from 'react-test-renderer'
import Presenter from './presenter'

test('header presenter', () => {
  const component = renderer.create(<Presenter />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
