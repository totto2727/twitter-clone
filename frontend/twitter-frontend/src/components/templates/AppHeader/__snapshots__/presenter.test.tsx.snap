// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header presenter 1`] = `
<header>
  header
</header>
`;
